File: python/1399-Count Largest Group.py
class Solution(object):
  def countLargestGroup(self, n):
    """
    :type n: int
    :rtype: int
    """
    myDict = {}
    thisSum = 0
    for i in range(1, n + 1):
      thisSum = 0
      # Get the sum of i's digits
      value = i
      thisSum += value % 10
      while value > 0:
        value = value / 10
        thisSum += value % 10
        
      if thisSum in myDict:
        myDict[thisSum] += 1
      else:
        myDict[thisSum] = 1
    
    maxCount = max(myDict.values())
    ret = 0
    # How many keys share this max value?
    for key, val in myDict.items():
      if val == maxCount:
        ret += 1
      print(key, val)
    
    return ret

File: cpp/1000 to 1999/1399-Count Largest Group.cpp
#include <vector>

class Solution
{
public:
  int countLargestGroup(int n)
  {
    int ret = 0;
    std::vector<int> sumToCount(37, 0); // 37 comes from the largest n allowed by constraints, 10^4 - 1, 9999, 9*4 = 36
    for(int i = 1; i <= n; i++)
    {
      int val = i;
      int thisSum = 0;
      while(val > 0)
      {
        thisSum += val % 10;
        val = val / 10;
      }
      sumToCount[thisSum]++;
    }

    int maxCnt = 0;
    for(const auto cnt : sumToCount)
    {
      maxCnt = std::max(maxCnt, cnt);
    }
    for(const auto cnt : sumToCount)
    {
      if(cnt == maxCnt)
      {
        ret++;
      }
    }
    return ret;
  }
};